qtconsole/tests/test_console_widget.py

```diff
@@ -172,4 +175,53 @@ def test_keypresses(self):
         self.assertEqual(control.document().findBlockByNumber(3).text(), '> line3')
 
         # TODO: many more keybindings
-        
+
+    def test_complete(self):
+        class TestKernelClient(object):
+            def is_complete(self, source):
+                calls.append(source)
+                return msg_id
+        w = ConsoleWidget()
+        cursor = w._get_prompt_cursor()
+        w._execute = lambda *args: calls.append(args)
+        w.kernel_client = TestKernelClient()
+        msg_id = object()
+        calls = []
+
+        # test incomplete statement (no _execute called, but indent added)
+        w.execute("thing", interactive=True)
+        self.assertEqual(calls, ["thing"])
+        calls = []
+        w._handle_is_complete_reply(
+            dict(parent_header=dict(msg_id=msg_id),
+                 content=dict(status="incomplete", indent="!!!")))
+        self.assert_text_equal(cursor, "thing\u2029> !!!")
```

Comment:
This test fails for me on Python 2.7 (PySide 1.2.2, Qt 4.8.7, OS X 10.12.5), on the result of merging this branch into master. Traceback:
```python
======================================================================
FAIL: test_complete (qtconsole.tests.test_console_widget.TestConsoleWidget)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mdickinson/Enthought/Projects/Canopy_Geoscience/qtconsole/qtconsole/tests/test_console_widget.py", line 198, in test_complete
    self.assert_text_equal(cursor, "thing\u2029> !!!")
  File "/Users/mdickinson/Enthought/Projects/Canopy_Geoscience/qtconsole/qtconsole/tests/test_console_widget.py", line 32, in assert_text_equal
    self.assertEqual(selection, text)
AssertionError: u'thing\u2029> !!!' != 'thing\\u2029> !!!'
```
It looks as though the fix is as simple as replacing the second argument to `self.assert_text_equal` with something that's explicitly Unicode: `u"thing\u2029> !!!"`.

Not sure why the test failure doesn't show up in the Travis CI runs, but I notice that those runs seem to run significantly fewer tests than I get by running `nosetests` at the prompt.